cloud/auth.go

```diff
@@ -562,6 +572,16 @@ func (c *Client) getSSHCredentials(challenge string, key *agent.Key) (credential
 	return credentials, nil
 }
 
+func (c *Client) maybeWarnAuthExpiration() {
+	if authExpirationWarningDisplayed || c.lastAuthMethodExpiry.IsZero() {
+		return
+	}
+	if diff := c.lastAuthMethodExpiry.Sub(time.Now().UTC()); diff >= 0 && diff <= authExpirationWarningGracePeriod {
```

Comment:
`diff` doesn't make it clear if it's the duration until or after expiry.

maybe call it something like `remaining` instead?

```suggestion
	if remaining := c.lastAuthMethodExpiry.Sub(time.Now().UTC()); remaining >= 0 && remaining <= authExpirationWarningGracePeriod {
```